ambry-api/src/test/java/com.github.ambry/rest/MockRestResponseChannel.java

```diff
@@ -132,8 +132,12 @@ public synchronized void onResponseComplete(Exception exception) {
       this.exception = exception;
       try {
         if (!responseMetadataFinalized.get() && exception != null) {
-          // clear headers
+          // clear headers except for the value of Allow
+          String allow = getHeader(RestUtils.Headers.ALLOW);
           responseMetadata.put(RESPONSE_HEADERS_KEY, new JSONObject());
+          if (allow != null && !allow.isEmpty()) {
```

Comment:
Minor: if we don't import guava which has nice utilities like Strings.isNullOrEmpty(), then would be good to implement our own somewhere.

Reply:
Done